docs/proposal/machine-config.md

```diff
@@ -0,0 +1,65 @@
+# AWS Provider Machine Configuration and Status
+
+## Overview
+This proposal aims to provide an initial API for AWS machine configuration and status for the AWS machine actuator.
+Cluster configuration for the AWS provider is out of scope for this proposal.
+
+## AWSMachineProviderConfig
+Defines the spec portion of a Machine specific to AWS. It is meant to be embedded in the
+Machine.Spec.ProviderConfig field of a cluster API Machine.
+
+Following are design decisions around this type:
+
+- No AWS SDK types are included. For the following reasons:
+  - Clients would also need to have the AWS SDK vendored.
+  - The type would be tied to a specific AWS client version.
```

Comment:
This 100% correct and valid, but we'll provide only subset of configurations, which you can do, when launching a machine.

For example in our project we decided to more or less copy the structs which were used in aws-client-go. https://github.com/gardener/machine-controller-manager/blob/5976512b001db0b029a8982dc93e1065f099e4a7/pkg/apis/machine/v1alpha1/types.go#L763-L885 . I'm not 100% sure that our approach is 100% valid as well.

Reply:
Can we adopt the approach of including as much of the API as we need for what we initially support and as the actuator evolves add more if we need to?

Reply:
I think this is really the best and most pragmatic choice.